tests/utils/state-locks-filter

```diff
@@ -0,0 +1,313 @@
+                self.traces[trace] = sorted(times, key=lambda x: x.get_held_time(), reverse=True)
+        if sort_wait_time:
+            for trace, times in self.traces.items():
+                self.traces[trace] = sorted(times, key=lambda x: x.get_held_time(), reverse=True)
+
+        for trace, times in self.traces.items():                
+            print("-"*20+"TRACE"+"-"*20)
+            print("")
+            print(trace)
+            print("")
+            for time in times:
+                print("{} held: {} ms, wait: {} ms".format(time.get_group_name(), time.get_held_time(), time.get_wait_time()))
+            print("")                
+
+class LocksFileReader:
+
+    PROJECT_PREFIX = "###START: SNAPD PROJECT"
+    TEST_PREFIX = "###START:"
+
+    def __init__(self, locks_file: str):
+        self.locks_file = locks_file
+        self.lines = list[str]()
+        self.groups = list[LocksGroup]()
+
+        self._read()
+
+    def _read(self):
+        with open(self.locks_file, "r") as f:
+            self.lines = f.readlines()
+
+        current_line = 0
+        if not self._is_project(self.lines[current_line]):
+            print("First time expected to be the project start.")
+            sys.exit(1)
+
+        # Read the tests
+        while True:
+            if current_line >= len(self.lines):
+                return
+
+            group_lines = self._current_group(current_line)
+            if len(group_lines) == 0:
+                print("Error parsing test.")
+                sys.exit(1)
+
+            self.groups.append(LocksGroup(group_lines))
+            current_line = current_line + len(group_lines)
+
+    # Indicates if the line is a test
+    def _is_test(self, line: str) -> bool:
+        return not self._is_project(line) and line.startswith(self.TEST_PREFIX)
+
+    # Indicates if the line is the project declaration
+    def _is_project(self, line: str) -> bool:
+        return line.startswith(self.PROJECT_PREFIX)
+    
+    # This function works to detect the current test but also for the current project
+    def _current_group(self, start_line: int) -> list[str]:
+        curr_test = []
+        curr_test.append(self.lines[start_line])
+
+        current_line = start_line + 1
+        while True:
+            if current_line >= len(self.lines) or self._is_test(self.lines[current_line]): 
+                return curr_test
+            
+            curr_test.append(self.lines[current_line])
+            current_line = current_line + 1
+
+    # Retrieve the test lines
+    def get_test(self, test: str) -> list[str]:
+        for group in self.groups:
+            if test in group.get_name():
+                return str(group)
+            
+        return ""
+
+    # Retrieve the times for each trace in the file
+    # For each trace, there is a list with the times for each test where the trace appears
+    def get_traces_times(self) -> dict[LockTrace, list[GroupTimes]]:
```

Comment:
this looks about right, but as I said I'm not quite sure without very carefull what is a group

Reply:
docs added now